test/whitelist/binutils/newlib.log

```diff
@@ -0,0 +1,9 @@
+#
+# -shared unsupported on bare-metal toolchain
+#
+FAIL: Build libpr23818.so
+FAIL: Build libpr23958.so
```

Comment:
Looks like these are just in the wrong list.  They should be in lto_link_elf_tests instead of lto_link_tests.  The lto_link_elf_tests list is only used when the target supports shared libraries and elf.  This is in ld/testsuite/ld-plugin/lto.exp.